From cc823ad298b7413874f0c3b369885ca2d645bdd5 Mon Sep 17 00:00:00 2001
From: Mike Gelfand <mikedld@mikedld.com>
Date: Tue, 8 Sep 2020 09:51:44 +0300
Subject: [PATCH] Include <stddef.h> to import size_t definition

This header is standard and is guaranteed to work even with C89/C90 which
is supported by any C compiler out there I could think of.
---
 include/b64/ccommon.h | 10 ----------
 include/b64/cdecode.h |  2 ++
 include/b64/cencode.h |  2 ++
 3 files changed, 4 insertions(+), 10 deletions(-)

diff --git a/include/b64/ccommon.h b/include/b64/ccommon.h
index 2b614df..74896b6 100644
--- a/include/b64/ccommon.h
+++ b/include/b64/ccommon.h
@@ -11,14 +11,4 @@ For details, see http://sourceforge.net/projects/libb64
 #define BASE64_VER_MAJOR	2
 #define BASE64_VER_MINOR	0
 
-#ifndef HAVE_SIZE_T
-  #ifdef _WIN32
-    #include <crtdefs.h>
-  #elseif defined (__unix__) || (defined (__APPLE__) && defined (__MACH__))
-    #include <stdlib.h>
-  #else
-    typedef unsigned long size_t;
-  #endif
-#endif
-
 #endif /* BASE64_CCOMMON_H */
diff --git a/include/b64/cdecode.h b/include/b64/cdecode.h
index d6ff24c..d9f00e3 100644
--- a/include/b64/cdecode.h
+++ b/include/b64/cdecode.h
@@ -10,6 +10,8 @@ For details, see http://sourceforge.net/projects/libb64
 
 #include "ccommon.h"
 
+#include <stddef.h>
+
 #define BASE64_CDEC_VER_MAJOR   2
 #define BASE64_CDEC_VER_MINOR   0
 
diff --git a/include/b64/cencode.h b/include/b64/cencode.h
index 96b0cdb..2edf00c 100644
--- a/include/b64/cencode.h
+++ b/include/b64/cencode.h
@@ -10,6 +10,8 @@ For details, see http://sourceforge.net/projects/libb64
 
 #include "ccommon.h"
 
+#include <stddef.h>
+
 #define BASE64_CENC_VER_MAJOR	2
 #define BASE64_CENC_VER_MINOR	0
 
-- 
2.45.1

